### PR TITLE
adds form fields for preferences to send compare route

### PR DESCRIPTION
**Pull request by dscottd7**
This should work with changes made to /compare route in the backend.

### CODE REVIEW

File: src/pages/HomePage.js [hunk 1]
```diff
@@ -1,80 +1,122 @@
...
+    if (name === 'url1' || name === 'url2') {
+      setUrls(prevUrls => ({
+        ...prevUrls,
+        [name]: e.target.value,
+      }));
+    } else if (name === 'selected_categories') {
+      const selectedOptions = Array.from(e.target.selectedOptions);
+      const selectedValues = selectedOptions.map(option => option.value);
+      setPreferences(prevPreferences => ({
+        ...prevPreferences,
```
Comment (edwmai) [inline src/pages/HomePage.js:33]: If the intention is to reset the preferences state without retaining previous values every time selected_categories is updated, then spreading ...prevPreferences would not be necessary. Excluding it ensures that preferences initializes to a clean state with no defaults whenever selected_categories changes

File: src/pages/HomePage.js [hunk 2]
```diff
@@ -1,80 +1,122 @@
...
+        selected_categories: selectedValues,
+      }));
+    } else {
+      setPreferences(prevPreferences => ({
+        ...prevPreferences,
```
Comment (edwmai) [inline src/pages/HomePage.js:38]: Same as line 33. Remove `...prevPreferences` to reset user preferences.

**Comment by jbh14** [comment]
@edwmai , are we sure we want Preferences for the search to reset every time the user changes something?

I found the way it's working now pretty intuitive - when doing multiple searches in a row, I might swap out one product after a given search, but I usually want to keep my categories/instructions the same (think use case - if you're looking to buy a specific thing, you likely care about the same factors during a series of searches, but might want to look at comparing a few different sets of products).

**Review by superhermione — COMMENTED** [review]
For now, it is working well with the backend. My suggestion is to style the select section and make them in "checkbox" style so that user can make choices in a more visual way. Now, user has to choose one or use keyboard to click "Ctrl" to multi select.

File: src/pages/HomePage.js (hunk 1 above)
Comment (dscottd7) [reply src/pages/HomePage.js:33]: I was treating these two new fields the same as the urls, which aren't reset after the submit `Compare` is pressed. My thinking was that a user may want to modify the comparison they just made. We could assume they don't want that, and reset the fields. Or we could provide a `Reset` button to clear the form.

**Comment by edwmai** [comment]
@jbh14 The categories/selected preferences will still be retain in between comparisons since it will only update onChange. The problem will arise when deselected a previously selected preference. Since we are spreading the preferences, previous preference values will still be selected even if a user deselected the preference.

**Review by edwmai — APPROVED** [review]

**Pull request merged**